### PR TITLE
WT-4961 Checkpoints with cache overflow must keep history for reads. (Clang Format migration)

### PR DESCRIPTION
**Pull request by tetsuo-cpp**
This is a migration of #4786.

### CODE REVIEW

File: src/btree/bt_sync.c [hunk 1]
```diff
@@ -299,21 +299,9 @@ __wt_sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
...
-             * readers older than the checkpoint.  Otherwise, a bug
-             * in eviction can mark the page clean and discard
-             * history, causing those reads to incorrectly see
-             * newer versions of data than they should.
              */
             if (!WT_PAGE_IS_INTERNAL(page) && page->read_gen == WT_READGEN_WONT_NEED &&
-              !tried_eviction && F_ISSET(&session->txn, WT_TXN_HAS_SNAPSHOT) &&
-              (!F_ISSET(txn, WT_TXN_HAS_TS_READ) ||
-                txn->read_timestamp == conn->txn_global.pinned_timestamp)) {
+              !tried_eviction) {
```
Comment (tetsuo-cpp) [inline src/btree/bt_sync.c]: This was the only conflict involved. Someone will need to take a look at this.

Comment (keithbostic) [reply src/btree/bt_sync.c]: @agorrod should confirm, but this looks correct to me, that is, the current PR branch looks right.

**Comment by lukech** [comment]
It looks `wiredtiger-test-format-stress-sanitizer` job is still running on the `wt-4961-checkpoint-eviction-keep-history` branch and failing periodically. While I'm not sure if it deserves more time to run the feature branch (considering there were known failures), after clang format merge and a new migration branch created, it makes sense at least to switch the branch over to the new one in the Jenkins job?

**Comment by keithbostic** [comment]
@agorrod, @bvpvamsikrishna, this branch ran successfully overnight with a mixture of `CONFIG.stress` and the CONFIG that was failing on my local machine, so if you approve it, I think we could merge.

We're going to need a lot of burn-in on this change, I think, before we let it get downstream.

**Comment by lukech** [comment]
FYI - I updated `wiredtiger-test-format-stress-sanitizer` job to test against `wt-4961-checkpoint-eviction-keep-history-migration` branch (used in this PR).

**Comment by tetsuo-cpp** [comment]
Is this PR just waiting on testing?

**Comment by agorrod** [comment]
@bvpvamsikrishna is working on diagnosing some Evergreen test failures, that appear to be related to the checkpoint timestamp being reset (at least, that's what I remember from the last standup).

**Comment by bvpvamsikrishna** [comment]
@agorrod, @keithbostic  could you review the last commit. Currently, there is one failure in the evergreen patch build with this change.

**Comment by agorrod** [comment]
Nice work figuring this failure out Vamsi - it's nice to see the Evergreen tests green.

I'm not very comfortable using 1 as a special timestamp value - I think we should move the code away from using a WT_TS_NONE value as an indicator and keep a separate variable to track that if we can.

**Comment by bvpvamsikrishna** [comment]
@agorrod , @keithbostic , I have reverted the if condition in rollback_to_stable_check as I think we can rollback even if the first time checkpoint is not complete.  request for a review.

**Comment by keithbostic** [comment]
@bvpvamsikrishna, can you please explain what it means when we rollback-to-stable if we don't have a checkpoint in the system? I guess I'm (at least), confused about what the word "stable" means in that situation. Regardless, I'll defer to you & @agorrod on that question.

Also, the changes to require a checkpoint before rollback-to-stable was allowed were more pervasive than this: there were changes to the documentation, example code and test code as well, and all that needs to be rolled back.

**Comment by bvpvamsikrishna** [comment]
> @bvpvamsikrishna, can you please explain what it means when we rollback-to-stable if we don't have a checkpoint in the system? I guess I'm (at least), confused about what the word "stable" means in that situation. Regardless, I'll defer to you & @agorrod on that question.

I am considering rollback to stable as "discarding data from cache overflow" rather than in relation with checkpoint. So, after the system started and before first checkpoint:
1. some operations are performed.
2. Stable timestamp is set.
3. some more operations are performed.
4. rollback to stable is invoked.
In this scenario, the data till stable timestamp to be retained.

**Comment by agorrod** [comment]
Let's talk this through when I'm back in Australia on Tuesday - I haven't been following along and I think it'll be easier in person.

**Comment by bvpvamsikrishna** [comment]
> Let's talk this through when I'm back in Australia on Tuesday - I haven't been following along and I think it'll be easier in person.

Sure.

File: src/btree/bt_sync.c (hunk 1 above)
Comment (agorrod) [reply src/btree/bt_sync.c]: @keithbostic good question about backing out the workaround. This is the change that does that, and it looks correct to me

**Comment by bvpvamsikrishna** [comment]
@agorrod , @keithbostic I have updated the branch with the changes we have discussed in the morning. This is link to [evergreen patch build](https://evergreen.mongodb.com/version/5d8ab0a72fbabe60735e120c).

Request your review.

**Pull request merged**